src/storage/disk_manager.c

```diff
@@ -1205,24 +1207,57 @@ disk_get_link (THREAD_ENTRY * thread_p, INT16 volid, INT16 * next_volid, char *n
   return next_volext_fullname;
 }
 
+/*
+ * disk_adapt_other_config_volume_full_name_to_local_config - adapt the full volume path and name
+ *                      coming, possibly, from an old or different config to the current config
+ */
+void
+disk_adapt_other_config_volume_full_name_to_local_config (char *local_config_full_name,
+							  const char *other_config_full_name)
+{
+  assert (local_config_full_name != nullptr);
+
+  // reconstruct a current-configuration-adapted volume full name
+  // by extracting the file name from the fully qualified name
+  const char *const vol_filename = fileio_get_base_file_name (other_config_full_name);
+  if (vol_filename == nullptr || vol_filename[0] == '\0')
+    {
+      local_config_full_name[0] = '\0';
+    }
+  else
+    {
+      // and the rest of the path from the current settings
+      char buf_temp_path[PATH_MAX];
+      const char *const vol_base_path = fileio_get_directory_path (buf_temp_path, boot_db_full_name ());
+
+      fileio_make_volume_ext_given_name (local_config_full_name, vol_base_path, vol_filename);
+    }
```

Comment:
With the assert above, this can be rewritten as:
```suggestion
  assert (vol_filename != nullptr && vol_filename[0] != '\0');
  // and the rest of the path from the current settings
  char buf_temp_path[PATH_MAX];
  const char *const vol_base_path = fileio_get_directory_path (buf_temp_path, boot_db_full_name ());

  fileio_make_volume_ext_given_name (local_config_full_name, vol_base_path, vol_filename);
```

Reply:
Function argument `other_config_full_name` cannot be null (because it comes from the volume header `var_fields` which are initialized to empty string) but it can be empty. Asserted accordingly.